src/cursor/cur_std.c

```diff
@@ -1176,42 +1193,52 @@ __wt_cursor_largest_key(WT_CURSOR *cursor)
 int
 __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CONFIG_ITEM cval;
     WT_CURSOR_BTREE *cbt;
     WT_DECL_RET;
     WT_ITEM key;
     WT_SESSION_IMPL *session;
     int exact;
-    bool inclusive;
+    char *cfg;
+    bool inclusive, have_action;
 
+    cfg = NULL;
     cbt = (WT_CURSOR_BTREE *)cursor;
     exact = 0;
-    inclusive = false;
+    inclusive = true;
+    have_action = false;
 
-    CURSOR_API_CALL_CONF(cursor, session, bound, config, cfg, NULL);
+    CURSOR_API_CALL(cursor, session, bound, NULL);
 
     if (CUR2BT(cursor)->type == BTREE_COL_FIX)
         WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with fixed column store.");
 
     if (F_ISSET(cursor, WT_CURSTD_PREFIX_SEARCH))
         WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with prefix search.");
 
-    WT_ERR(__wt_config_gets(session, cfg, "action", &cval));
-    if (WT_STRING_MATCH("set", cval.str, cval.len)) {
-        WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
-        inclusive = cval.val != 0;
+    if (config == NULL || config[0] == '\0')
```

Comment:
What happens if the config is not "action"? does that mean we just no-op?

Reply:
The `have_action` bool handles cases where a user supplies a config such as "bound=upper" with no "action=set". In that case `have_action` will be false and we enter the bound setting if statement on line 1225. The reason we needed `have_action` is that we specified that "set" was the default action. This was okay while the WiredTiger config parsing framework conflated the configuration with the defaults but without that we have to pretend we did the same. This sort of answers your question in https://github.com/wiredtiger/wiredtiger/pull/8342#discussion_r993962883

Reply:
I will add a comment explaining the boolean a bit better.

Reply:
e0340ab10